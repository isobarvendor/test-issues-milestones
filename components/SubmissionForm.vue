<template>
  <div>
    <div>
      <div class="img-footer">
        <img src="/img/landing/instruments.png" width="100%" />
      </div>
      <div class="header">
        {{
          form.name
            ? submissionText.hello + " " + form.name
            : submissionText.header
        }}
      </div>
      <form class="mechanics" autocomplete="off">
        <div
          class="details"
          v-if="submissionFormFields && submissionFormFields.isNameActive"
        >
          <input
            id="name"
            type="text"
            name="name"
            v-model="form.name"
            v-validate="'required'"
            :placeholder="submissionText.name"
          />
          <div style="text-align: left;">
            <p class="error-message">
              {{ errors.first("name") ? submissionText.errorRequiredName : "" }}
            </p>
            <p class="name-tip">{{ submissionText.nameTip }}</p>
          </div>
        </div>
        <div
          class="details"
          v-if="submissionFormFields && submissionFormFields.isEmailActive"
        >
          <input
            id="email"
            type="email"
            name="email"
            v-model="form.email"
            v-validate="'required'"
            :placeholder="submissionText.email"
            readonly
          />
          <!--span class="error-message">{{ errors.first('email') }}</span-->
        </div>
        <div
          class="details"
          v-if="
            submissionFormFields && submissionFormFields.isPhoneNumberActive
          "
        >
          <div class="btn-text">
            <input
              id="phoneCode"
              type="tel"
              v-model="phoneCodeDisplay"
              :readonly="true"
            />
            <input
              id="phoneNumber"
              type="tel"
              name="phone"
              v-model="form.phoneNumber"
              v-validate="'required'"
              class="short"
              :placeholder="submissionText.phoneNumber"
            />
          </div>
          <div class="info-icon tooltip">
            <img src="/img/landing/info-button.png" width="25" />
            <span class="tooltiptext">{{ submissionText.phoneTooltip }}</span>
          </div>
          <span class="error-message">{{
            errors.first("phoneNumber")
              ? errors.first("phoneNumber").includes("required")
                ? submissionText.errorRequiredPhone
                : errors.first("phoneNumber")
              : ""
          }}</span>
        </div>

        <!--div v-if="submissionType=='with_receipt'" class="details receipt">
      <div v-if="!image">

       <label for="file-upload" class="custom-file-upload">
       <img src="/img/icons/upload-icon.png"/> <span class="labels">Upload receipt</span>
      </label>
       <input id="file-upload" type="file" @change="onFileChange" value="uploadReceipt">
      </div>
      <div v-else>
        <img :src="image" width="100" />
        <button @click="removeImage">Remove image</button>
      </div>
    </div-->

        <br />

        <div class="row top">
          <div class="col d-flex consent">
            <div class="checkbox">
              <label for="form_tnc">
                <input
                  type="checkbox"
                  name="tnc"
                  id="form_tnc"
                  v-model="form.terms"
                />
                <span></span>
              </label>
            </div>
            <div class="terms" v-html="submissionText.acceptTerm"></div>
          </div>
        </div>
        <div class="row top">
          <div class="col d-flex consent">
            <div class="checkbox">
              <label for="form_age">
                <input
                  type="checkbox"
                  name="ageConsent"
                  id="form_age"
                  v-model="form.ageConsent"
                />
                <span></span>
              </label>
            </div>
            <div class="terms" v-html="submissionText.declareAge"></div>
          </div>
        </div>
        <div class="row top">
          <div class="col d-flex consent">
            <div class="checkbox">
              <label for="form_pp">
                <input
                  type="checkbox"
                  name="privacy"
                  id="form_pp"
                  v-model="form.privacy"
                />
                <span></span>
              </label>
            </div>
            <div class="terms" v-html="submissionText.acceptPrivacy"></div>
          </div>
        </div>

        <div
          class="error-message-black"
          v-if="errorMessage"
          v-html="errorMessage"
        ></div>
        <div class="btn-area">
          <div class="info-btn">
            <div class="btn-text">
              <input
                id="code"
                v-model="form.code"
                v-validate="'required'"
                type="text"
                name="code"
                :placeholder="submissionText.enterCode"
              />
              <span class="error-message-red">{{
                errors.first("code")
                  ? submissionText.theCodeFieldIsRequired
                  : null
              }}</span>
            </div>
            <div class="info-icon tooltip">
              <img src="/img/landing/info-button.png" width="25" />
              <span class="tooltiptext">{{ submissionText.tooltipText }}</span>
            </div>
          </div>
          <div style="padding:20px" v-if="loading">
            <v-progress-circular
              :width="2"
              color="white"
              indeterminate
            ></v-progress-circular>
          </div>
          <v-btn
            class="get-code"
            id="form_submission"
            dark
            v-else
            v-on:click="submit()"
            >{{ submissionText.buttonText }}</v-btn
          >
        </div>
      </form>
    </div>

    <!--div v-else class="thanks"-->

    <!--div v-if="prizeWin.allocationArray">
        <div class="header">{{thankyouPage.Title}}</div>
        <div>{{  prizeWin.allocationArray[0].amount }}&nbsp;
            {{thankyouPage.Message}}
        </div>
        <div>
          <v-btn class="get-code" v-on:click="goToRewards()">View Rewards</v-btn>
        </div>
    </div>
     <div v-if="prizeWin.participationInserted">
           <div class="header" >{{thankyouSubmission.Ttitle}} </div>

        <div>
            {{thankyouSubmission.Message}} <a href='#'>{{form.email}}</a>
        </div>
    </div>
    <div v-if="prizeWin.instantWinResult&&prizeWin.instantWinResult.redeemedPrize.status=='claimed'">
        <div class="header">{{thankyouPage.Title}}</div>
        <div class="header">{{prizeWin.instantWinResult.redeemedPrize.name}} </div>
        <div>
            <img :src="prizeWin.instantWinResult.redeemedPrize.imgUrl" width="250" />
        </div>
        <div>
            {{prizeWin.instantWinResult.redeemedPrize.shortDescription}}
        </div>
    </div>
    <div v-else-if="prizeWin.instantWinResult&&prizeWin.instantWinResult.redeemedPrize.status!='claimed'">
      <div class="header" >{{thankyouSubmission.Ttitle}} </div>

       <div>
          {{prizeWin.instantWinResult.redeemedPrize.redeemDescription}}
       </div>
      <div>
          {{thankyouSubmission.Message}} <a href='#'>{{form.email}}</a>
      </div>
    </div>
  </div-->
  </div>
</template>

<script>
import {
  CHECK_MIXCODE,
  CHECK_ATTEMPT,
  GET_LIST_WALLET,
  GET_USER_DATA
} from "@/store/action_types";
import { translation } from "@/constants/index";
import * as _ from "lodash";
export default {
  name: "Form",
  inject: ["$validator"],
  props: {
    data: null,
    cmsData: null
  },
  data() {
    return {
      form: {
        name: null,
        email: null,
        code: null,
        terms: false,
        privacy: false,
        uploadFile: null,
        phoneNumber: null,
        ageConsent: null
      },
      errorMessage: null,
      submitted: true,
      image: "",
      amazonImage: "",
      loading: false,
      prizeWin: null,
      phoneCodeDisplay: "+" + this.$config.phoneCode,
      phoneCode: this.$config.phoneCode + "-",
      showPhone: false,
      submissionText: translation.submissionText
    };
  },
  computed: {
    /* submissionType(){

      return this.data.campaignTypes.submissionType;
    },*/
    submissionFormFields() {
      return this.data.submissionFormFields;
    },
    campaignType() {
      return "";
      //return this.data.campaignTypes.mechanicType;
    },

    maxPhoneNumber() {
      return this.$config.maxPhoneNumber;
    },

    /*campaignTitle(){
      return this.data.campaignTypes.Title;
    },*/

    thankyouSubmission() {
      return this.cmsData.ThankYouSubmission;
    },
    thankyouPage() {
      return this.cmsData.ThanksYouPage;
    },
    loginInfo() {
      return this.$store.getters.getLoginAccount;
    },
    getAttempt() {
      return this.data.attempts;
    }
  },
  methods: {
    generateRequest(currentAttempt) {
      if (currentAttempt >= this.getAttempt.length) {
        currentAttempt = this.getAttempt.length - 1;
      }

      let mixCode = this.getAttempt[currentAttempt].mixCode;
      let ngps = this.getAttempt[currentAttempt].NPGS;
      let programId = null;
      this.attemptData = this.getAttempt[currentAttempt];

      if (mixCode.length > 0) {
        // console.log(mixCode);
        let programs = _.filter(mixCode, a => {
          return (
            a.codeInitial !== null &&
            a.codeInitial.toUpperCase() ==
              this.form.code.charAt(0).toUpperCase() &&
            a.characterLimit == this.form.code.length
          );
        });
        //console.log(programs);
        let programsNull = _.filter(mixCode, a => {
          return (
            (a.codeInitial == null || a.codeInitial == "") &&
            a.characterLimit == this.form.code.length
          );
        });
        if (programs.length > 0) {
          programId = programs[0].ProgrammeID;
        } else {
          programId =
            programsNull.length > 0 ? programsNull[0].ProgrammeID : null;
        }
      }

      if (!programId) {
        return false;
      }
      let request;
      request = {
        name: this.form.name,
        email: this.form.email,
        mechanic: this.getAttempt[currentAttempt].campaignType,
        programmeId: programId,
        configurationId: ngps[0].configID,
        flowLabel: ngps[0].flowLabel,
        termsAgreement: this.form.terms,
        privacyAgreement: this.form.privacy,
        ageAgreement: this.form.ageConsent
      };
      if (this.loginInfo) {
        //request["userId"]=this.loginInfo.uuid;
      }
      if (this.form.phoneNumber) {
        request["phone"] = this.phoneCode + this.form.phoneNumber;
      }
      if (this.form.code) {
        request["pin"] = this.form.code;
      }
      return request;
    },
    async checkcurrentAttempt() {
      await this.$store
        .dispatch(GET_USER_DATA)
        .then(response => {
          this.form.phoneNumber = response.data.phoneNumber
            .replace(this.phoneCode, "")
            .replace(this.phoneCodeDisplay, "");
          // this.showPhone = true;
          this.currentAttempt = response.data.currentAttemptNumber;
          this.form.terms = response.data.termsAgreement;
          this.form.privacy = response.data.privacyAgreement;
          this.form.ageConsent = response.data.ageAgreement;
        })
        .catch(error => {
          if (error) {
            this.currentAttempt = 9999999;
          }
        });
    },

    async submit() {
      let request = null;
      this.loading = true;
      let index = 0;
      let response = {};

      this.$validator.validateAll().then(async valid => {
        if (valid && this.errors.all().length <= 0) {
          let currentattempt = 0;
          if (!this.form.terms) {
            this.loading = false;
            this.errorMessage = this.submissionText.errorTerm;
            return false;
          }
          if (!this.form.privacy) {
            this.loading = false;
            this.errorMessage = this.submissionText.errorPolicy;
            return false;
          }
          if (!this.form.ageConsent) {
            this.loading = false;
            this.errorMessage = this.submissionText.errorDeclare;
            return false;
          }
          this.errorMessage = null;
          await this.checkcurrentAttempt();

          if (this.getAttempt) {
            //my code for submit
            request = this.generateRequest(this.currentAttempt);
            if (!request) {
              this.loading = false;
              this.errorMessage = this.submissionText.errorPinCode;
              return false;
            }
            const captchaResponse = await this.$recaptcha.execute("validation");
            request.captchaResponse = captchaResponse;
            this.$store
              .dispatch(CHECK_MIXCODE, request)
              .then(response => {
                // this.submitted=true;
                let loginData = {
                  ...this.$store.state.login,
                  phone: this.phoneCode,
                  name: this.form.name
                };
                // console.log(loginData);

                this.$store.commit("SET_LOGIN_ACCOUNT", loginData);
                this.loading = false;
                // console.log(response);
                let result = response.data;
                if (result.redeemed) {
                  this.errorMessage = this.submissionText.errorPinCode;
                }
                if (!result.redeemed) {
                  this.prizeWin = result;
                  let attemptData = this.attemptData;
                  let data = {
                    attemptData,
                    response: result,
                    request
                  };
                  //console.log(data);
                  this.$emit("submit", data);
                }
              })
              .catch(error => {
                this.loading = false;
                if (error.response) {
                  this.errorMessage = this.submissionText.errorAPI;
                }
                if (error.response && error.response.data.status == "400") {
                  this.errorMessage = this.submissionText.errorPinCode;
                }

                if (error.response && error.response.data.status == "401") {
                  localStorage.clear();
                  this.$store.commit("SET_LOGIN_ACCOUNT", null);
                  this.$store.commit("SET_TOKEN", null);
                  location.reload();
                }
              });
          } else {
            this.loading = false;
            this.errorMessage = this.submissionText.errorPinCode;
          }
        } else {
          this.loading = false;
        }
      });
    },
    async getAccount() {
      if (this.loginInfo) {
        this.form.name = this.loginInfo.name;
        this.form.email = this.loginInfo.email;
        if (this.loginInfo.phone) {
          this.form.phoneNumber = this.loginInfo.phone
            .replace(this.phoneCode, "")
            .replace(this.phoneCodeDisplay, "");
          this.showPhone = true;
        }
      }
      await this.checkcurrentAttempt();
      if (this.currentAttempt > 1) {
      }
    },

    getListWallet() {
      this.$store
        .dispatch(GET_LIST_WALLET)
        .then(response => {})
        .catch(error => {
          if (error.response && error.response.data.status == "401") {
            localStorage.clear();
          }
        });
    },
    goToRewards() {
      location.href = "/rewards";
    },
    addPhoneCode() {
      this.showPhone = true;
    }
  },
  beforeMount() {},
  async mounted() {
    try {
      await this.getAccount();
    } catch (e) {
      console.error(e);
    }
  },

  watch: {
    "form.phoneNumber": function(val) {
      let envs = this.$config;
      if (val.length == 1) {
        if (val == "0") {
          this.form.phoneNumber = "";
        }
      }
      if (
        val.length > envs.maxPhoneNumber ||
        val.length < envs.minPhoneNumber
      ) {
        this.errors.clear();
        this.$validator.errors.add({
          field: "phoneNumber",
          msg: "តម្រូវឲ្យមានលេខទូរស័ព្ទ"
        });
      } else if (isNaN(val)) {
        this.errors.clear();
        this.$validator.errors.add({
          field: "phoneNumber",
          msg: "តម្រូវឲ្យមានលេខទូរស័ព្ទ"
        });
      } else {
        this.errors.clear();
      }
    }
  }
};
</script>

<style scoped>
.details {
  position: relative;
}
.error-message-red {
  color: red;
}
.error-message {
  color: #000;
  text-align: left;
}
.error-message-black {
  color: #000;
}

.d-flex {
  display: flex;
  align-items: center;
}
.get-code {
  font-family: "Koulen-Regular" !important;
  display: block;
  width: 100%;
  margin: 25px 0;
}
input[type="file"] {
  display: none;
}
.custom-file-upload {
  border: 1px solid #ccc;
  display: inline-block;
  padding: 12px 12px;
  cursor: pointer;
  width: 100%;
  font-size: 14px;
}
.custom-file-upload span {
  padding-left: 20px;
}
.btn-area {
  text-align: center;
  margin-top: 30px;
}
button.get-code {
  height: 60px !important;
  border-radius: 15px;
}

form.mechanics {
  margin: 0 1em;
}
.thanks {
  text-align: center;
}
.info-btn {
  background: #fff;
  padding: 20px;
  border-radius: 15px;
  cursor: pointer;
  position: relative;
}
.btn-text {
  color: #000;
  text-decoration: #000;
}
.info-btn .info-icon {
  position: absolute;
  right: 10px;
  top: 34%;
}
.details .info-icon {
  position: absolute;
  right: 10px;
  top: 0px;
}
form input#code::placeholder {
  color: #1d1d1b;
  text-decoration: underline;
}
form input#code {
  color: #1d1d1b;
  text-align: center;
}
.tooltip {
  display: inline-block;
}
@media only screen and (min-width: 769px) {
  .tooltip .tooltiptext {
    visibility: hidden;
    width: 280px;
    background-color: #1d1d1b;
    color: #fff;
    border-radius: 6px;
    padding: 15px;
    position: absolute;
    z-index: 1;
    left: 150%;
    top: -15px;
    text-align: left;
  }
  .info-btn .tooltip .tooltiptext {
    top: -35px;
  }
  .details .tooltip .tooltiptext {
    top: -40px;
  }

  .tooltip .tooltiptext::after {
    content: "";
    position: absolute;
    top: 50%;
    right: 100%;
    margin-top: -10px;
    border-width: 10px;
    border-style: solid;
    border-color: transparent #1d1d1b transparent transparent;
  }
}
.tooltip:hover .tooltiptext {
  visibility: visible;
}
@media only screen and (max-width: 768px) {
  .tooltip .tooltiptext {
    visibility: hidden;
    width: 280px;
    background-color: #1d1d1b;
    color: #fff;
    text-align: left;
    border-radius: 6px;
    padding: 15px;
    position: absolute;
    z-index: 1;
    top: 150%;
    left: -720%;
    margin-left: -60px;
  }

  .tooltip .tooltiptext::after {
    content: "";
    position: absolute;
    bottom: 100%;
    left: 90%;
    margin-left: -10px;
    border-width: 10px;
    border-style: solid;
    border-color: transparent transparent #1d1d1b transparent;
  }
}
form input#phoneCode {
  width: 45px !important;
  display: inline-block;
  border: none;
}
form input#phoneNumber {
  border: none;
}
form input#phoneNumber.short {
  width: calc(100% - 65px) !important;
  border: none;
}
.details .btn-text {
  border-bottom: solid 1px #fff;
}
form.mechanics {
  margin: auto;
}
</style>
