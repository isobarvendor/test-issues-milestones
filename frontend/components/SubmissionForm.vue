<template>
  <form>

    <div>
      <input type="text" name="name" placeholder="Name"/>
    </div>
    <div>
      <input type="email" name="email" placeholder="Email"/>
    </div>
    <div>
      <input type="number" name="code" placeholder="10-digit code"/>
    </div>

    <br>

    <div class="row top">
      <div class="col d-flex">
        <div class="checkbox">
          <label for="form_tnc">
            <input type="checkbox" name="tnc" id="form_tnc">
            <span></span>
          </label>
        </div>
        <div>I accept the <a href="">Terms and Conditions</a> of this this redemption.</div>
      </div>
    </div>
    
    <div class="row top">
      <div class="col d-flex">
        <div class="checkbox">
          <label for="form_pp">
            <input type="checkbox" name="privacy" id="form_pp">
            <span></span>
          </label>
        </div>
        <div>I accept the <a href="">Privacy Policy</a> of this redemption.</div>
      </div>
    </div>

    <v-btn class="get-code">Get code</v-btn>

  </form>    
</template>

<script>
export default {
    name:"Form"
}
</script>

<style scoped>
  .d-flex {
    display: flex;
    align-items: center;
  }
  .get-code {
    display: block;
    width: 100%;
    margin: 25px 0;
  }
</style>