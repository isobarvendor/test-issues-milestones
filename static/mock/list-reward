[ {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   }
]
