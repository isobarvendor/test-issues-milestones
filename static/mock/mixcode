{
                      "programId": 453269,
                      "code": "3WTD90GE5",
                      "lot": {
                        "lotId": 4455843436,
                        "lotName": "Coke 500ml PET",
                        "activateDate": 1611208800000,
                        "inactivateDate": 1627603200000,
                        "active": true,
                        "expired": false,
                        "bevProdPkg": {
                          "trademarkCd": "05",
                          "trademarkName": "Coca-Cola",
                          "brandCd": "001",
                          "brandName": "Coca-Cola",
                          "bevProdCd": "0001",
                          "bevProdName": "Coca-Cola",
                          "caffeinated": true,
                          "carbonated": true,
                          "calorieCategory": {
                            "code": 1,
                            "description": "Regular"
                          }
                        },
                        "releaseId": 494494,
                        "releaseName": "Coke PET",
                        "organizationId": 157168,
                        "organizationName": "Harish",
                        "realtimeCode": false,
                        "testRelease": false,
                        "pointValue": 0,
                        "bevProdOptions": []
                      },
                      "redeemed": false,
                      "actualCode": "3WTD90GE5"
                    }
