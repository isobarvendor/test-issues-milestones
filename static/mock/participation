{"id":16,"questionId":1,"answerId":3,"participationId":17}
