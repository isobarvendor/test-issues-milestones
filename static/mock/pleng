{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p>3 months subscription to Pleng Pro</p>",
      "shortDescription": "<p>ឈ្នះថ្លៃសេវា Pleng Pro Subscription រយៈពេល 3 ខែ</p>",
      "priority": 1,
      "prizeId": "ks8o4ixh",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8o4ixh/CAM prizes — Pleng Pro@2x.png",
      "name": "3 months subscription to Pleng Pro ",
      "redemptionLink": "https://pleng.onelink.me/pp89?pid=ddlink&af_web_dp=https%3A%2F%2Fwww.smart.com.kh%2Fkm%2Fpleng%2F&af_dp=plengmusic%3A%2F% 2F&af_partner_name=cc",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះ Pleng Pro Subscription រយៈពេល ៣ ខែ! រីករាយនូវការស្តាប់ចម្រៀងឥតដែនកំណត់។ សូមដោនឡូតកម្មវិធី Pleng ពី Smart ដើម្បីបញ្ចូលលេខកូដ និងដាក់ដំណើរការ Smart Pleng Pro។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ ២០២១។</p>",
      "voucherCode": "CNXFUUAKAY"
      
    }
  }
}