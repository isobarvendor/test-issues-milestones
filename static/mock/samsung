{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p></p>",
      "shortDescription": "<p>ទូរស័ព្ទសាមសុង Galaxy Note20 Ultra</p>",
      "priority": 1,
      "prizeId": "ks8ns5hv",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8ns5hv/1.CAM prize-N20 Ultra@2x.png",
      "totalAmount": 20,
      "minAge": 13,
      "name": "Samsung Galaxy Note20 Ultra",
      "redemptionLink": "",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះទូរស័ព្ទសាមសុង Galaxy Note20 Ultra! សូមទាក់ទងទៅលេខ 1-800-20-2222 (មិនគិតថ្លៃសេវាទូរស័ព្ទ ២៤/៧) ដើម្បីដឹងពីរបៀបប្តូរយករង្វាន់។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ</p>",
      "voucherCode": "CNXFUUAKAY"
    }
  }
}