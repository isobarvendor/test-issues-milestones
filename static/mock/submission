{
            "burnResult": [
                {
                    "pincode": "0J6TC2VP5",
                    "programId": "453269",
                    "lotId": "445580",
                    "burned": true
                }
            ],
            "participationId": "1ylazdy9kkxmpa53",
            "instantWinResult": {
                "winner": true,
              "redeemedPrize": {
                    "prizeId": "kkqhy9so",
                    "voucherCode": "Voucher891",
                    "status": "claimed",
                    "expiryDate": 1620018590000,
                    "name": "JOOX VIP PASS",
                    "description":"Conglratulations",
                    "shortDescription": "http://google.com",
                    "redeemDescription": "http://abcd",
                    "imgUrl": "/img/landing/week 1 prize.png",
                    "barcodeType": 1,
                    "emailSent": false,
                    "emailMessage": "Email could not be sent"
                }
            }
        }
