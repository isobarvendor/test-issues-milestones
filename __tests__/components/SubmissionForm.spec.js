
import { shallowMount,createLocalVue } from '@vue/test-utils'

import {mockCMSConfig,mockCMSContent} from "@/__mock__/mock";

import SubmissionForm from "@/components/SubmissionForm";

import Vuex from 'vuex';

const $route = {
  name: "register"
}
const event = {
  target: {
    files: [
      {
        name: 'image.png',
        size: 5000,
        type: 'image/png',

      },
    ],
    value:'image.png'
  },
}
let localVue, store;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    store = new Vuex.Store({
      getters: {
        getCMSContent() {
          return mockCMSContent;
        },
        getLoginAccount(){
          return {
            name:"ron",
            email:"ron@dsw.com",
            imageUrl:"tes"
          }
        },
        getCMSConfig(){
          return mockCMSConfig[0];
        }
      },
      state:{
        language: "id",
        listWallet:{
          walletStatus:[
            {amount :30}
          ]
        }
      }
    });
  });

const factory = () => {
  return shallowMount(SubmissionForm, {
    localVue, store,
    propsData: {
      cmsData:store.getters.getCMSContent[0],
      data :store.getters.getCMSConfig
    }
  });
};

describe("SubmissionForm", () => {

  test('makes a call to persist the image on image upload', async() => {
    // Mount the component
    const wrapper = factory();


    // Mock FileReader.readAsDataURL() to be a function that returns null
    const fileReaderSpy = jest.spyOn(FileReader.prototype, 'readAsDataURL').mockImplementation(() => null)

    // Spy on the component’s persist() method
    const persistSpy = jest.spyOn(wrapper.vm, 'createImage')


    // Manually trigger the component’s onFileChange() method
    await wrapper.vm.onFileChange(event)

    // Assert that the FileReader object was called with the uploaded image
    expect(fileReaderSpy).toHaveBeenCalledWith(event.target.files[0])

    // Assert that the component’s persist() method was called with the uploaded image
    expect(persistSpy).toHaveBeenCalledWith(event.target.files[0])

   expect(wrapper.vm.errorMessage).toBe(null);
     // Assert that the component’s persist() method was called with the uploaded image
     expect(wrapper.vm.form.uploadFile).toBe(event.target.files[0])



  })

  test("user type all correct data and submit", async() => {
    const wrapper = factory();
    const name = "ronald"
    const email = 'ronald.pranata@isobar.com';
    const privacy = true;

    const terms = true;

    wrapper.find('#name').setValue(name);
    wrapper.find('#email').setValue(email);

    wrapper.find('#form_tnc').setChecked(terms);
    wrapper.find('#form_pp').setChecked(privacy);

    expect(wrapper.vm.form.name).toBe(name);
    expect(wrapper.vm.form.email).toBe(email);

    expect(wrapper.vm.form.terms).toBe(terms);
    expect(wrapper.vm.form.privacy).toBe(privacy);

    await wrapper.find('.get-code').trigger('click');
    wrapper.vm.$validator.validate().then(()=>{
    expect(wrapper.vm.$validator.errors.any()).toBe(false);
    }).catch(()=>{})

  });

  test("user type no name data and submit", async() => {
    const wrapper = factory();
    const name = ""
    const email = 'ronald.pranata@isobar.com';
    const privacy = true;

    const terms = true;

    wrapper.find('#name').setValue(name);
    wrapper.find('#email').setValue(email);

    wrapper.find('#form_tnc').setChecked(terms);
    wrapper.find('#form_pp').setChecked(privacy);

    expect(wrapper.vm.form.name).toBe(name);
    expect(wrapper.vm.form.email).toBe(email);

    expect(wrapper.vm.form.terms).toBe(terms);
    expect(wrapper.vm.form.privacy).toBe(privacy);

    await wrapper.find('.get-code').trigger('click');

    wrapper.vm.$validator.validate().then(()=>{
      expect(wrapper.vm.loading).toBe(false);
      expect(wrapper.vm.$validator.errors.any()).toBe(true);
    }).catch(()=>{})

  });

  test("user not check terms", async() => {
    const wrapper = factory();
    const name = "ronald"
    const email = 'ronald.pranata@isobar.com';
    const privacy = true;

    const terms = true;

    wrapper.find('#name').setValue(name);
    wrapper.find('#email').setValue(email);


    wrapper.find('#form_pp').setChecked(privacy);

    expect(wrapper.vm.form.name).toBe(name);
    expect(wrapper.vm.form.email).toBe(email);

    expect(wrapper.vm.form.terms).toBe(false);
    expect(wrapper.vm.form.privacy).toBe(privacy);

    await wrapper.find('.get-code').trigger('click');

    wrapper.vm.$validator.validate().then(()=>{
      expect(wrapper.vm.loading).toBe(false);
      expect(wrapper.vm.errorMessage).toBe("accept our terms and conditions");
    }).catch(()=>{})

  });

  test("user not check privacy", async() => {
    const wrapper = factory();
    const name = "ronald"
    const email = 'ronald.pranata@isobar.com';
    const privacy = true;

    const terms = true;

    wrapper.find('#name').setValue(name);
    wrapper.find('#email').setValue(email);


    wrapper.find('#form_pp').setChecked(privacy);

    expect(wrapper.vm.form.name).toBe(name);
    expect(wrapper.vm.form.email).toBe(email);

    expect(wrapper.vm.form.terms).toBe(false);
    expect(wrapper.vm.form.privacy).toBe(privacy);

    await wrapper.find('.get-code').trigger('click');

    wrapper.vm.$validator.validate().then(()=>{
      expect(wrapper.vm.loading).toBe(false);
      expect(wrapper.vm.errorMessage).toBe("Please accept our privacy policies");
    }).catch(()=>{})

  });


  test("test all computed value", () => {
    const wrapper = factory();
    expect(wrapper.vm.submissionType).toBe(wrapper.vm.data.campaignTypes.submissionType);
    expect(wrapper.vm.submissionFormFields).toBe(wrapper.vm.data.submissionFormFields);
    expect(wrapper.vm.campaignType).toBe(wrapper.vm.data.campaignTypes.mechanicType);
    expect(wrapper.vm.campaignTitle).toBe(wrapper.vm.data.campaignTypes.Title);
    expect(wrapper.vm.thankyouSubmission).toBe(wrapper.vm.cmsData.ThankYouSubmission);
    expect(wrapper.vm.thankyouPage).toBe(wrapper.vm.cmsData.ThanksYouPage);
    expect(wrapper.vm.loginInfo).toBe(store.getters.getLoginAccount);

  });
  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
