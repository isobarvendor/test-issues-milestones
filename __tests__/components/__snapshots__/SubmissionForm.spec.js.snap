// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionForm renders properly 1`] = `
<div>
  <div class="header">Be the lucky one</div>
  <form autocomplete="off" class="mechanics">
    <div class="details"><input id="name" type="text" name="name" placeholder="Name" aria-required="true" aria-invalid="false"> <span class="error-message"></span></div>
    <div class="details"><input id="email" type="email" name="email" placeholder="Email" aria-required="true" aria-invalid="false"> <span class="error-message"></span></div>
    <div class="details"><input id="code" type="text" name="code" placeholder="10-digit code" aria-required="true" aria-invalid="false"> <span class="error-message"></span></div>
    <!----> <br>
    <div class="row top">
      <div class="col d-flex consent">
        <div class="checkbox"><label for="form_tnc"><input type="checkbox" name="tnc" id="form_tnc"> <span></span></label></div>
        <div class="terms">I accept the <a href="">Terms and Conditions</a> of this this redemption.</div>
      </div>
    </div>
    <div class="row top">
      <div class="col d-flex consent">
        <div class="checkbox"><label for="form_pp"><input type="checkbox" name="privacy" id="form_pp"> <span></span></label></div>
        <div class="terms">I accept the <a href="">Privacy Policy</a> of this redemption.</div>
      </div>
    </div>
    <!---->
    <div class="btn-area">
      <v-btn-stub tag="button" activeclass="" type="button" class="get-code">Submit</v-btn-stub>
    </div>
  </form>
</div>
`;
